<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from hydra.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="hydra" xmlns:xacro="http://ros.org/wiki/xacro">
  <!--
  Author: Kelsey Hawkins
  Contributers: Jimmy Da Silva, Ajit Krisshna N L, Muhammad Asif Rana
-->
  <!-- measured from model -->
  <link name="placer_base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125"/>
    </inertial>
  </link>
  <joint name="placer_shoulder_pan_joint" type="revolute">
    <parent link="placer_base_link"/>
    <child link="placer_shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1273"/>
    <axis xyz="0 0 1"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="placer_shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="7.778"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0314743125769" ixy="0.0" ixz="0.0" iyy="0.0314743125769" iyz="0.0" izz="0.021875625"/>
    </inertial>
  </link>
  <joint name="placer_shoulder_lift_joint" type="revolute">
    <parent link="placer_shoulder_link"/>
    <child link="placer_upper_arm_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.220941 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="placer_upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="12.93"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.306"/>
      <inertia ixx="0.421753803798" ixy="0.0" ixz="0.0" iyy="0.421753803798" iyz="0.0" izz="0.036365625"/>
    </inertial>
  </link>
  <joint name="placer_elbow_joint" type="revolute">
    <parent link="placer_upper_arm_link"/>
    <child link="placer_forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1719 0.612"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="placer_forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.87"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28615"/>
      <inertia ixx="0.111069694097" ixy="0.0" ixz="0.0" iyy="0.111069694097" iyz="0.0" izz="0.010884375"/>
    </inertial>
  </link>
  <joint name="placer_wrist_1_joint" type="revolute">
    <parent link="placer_forearm_link"/>
    <child link="placer_wrist_1_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.5723"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="placer_wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="placer_wrist_2_joint" type="revolute">
    <parent link="placer_wrist_1_link"/>
    <child link="placer_wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.1149 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="placer_wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="placer_wrist_3_joint" type="revolute">
    <parent link="placer_wrist_2_link"/>
    <child link="placer_wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1157"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="placer_wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.202"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000526462289415" ixy="0.0" ixz="0.0" iyy="0.000526462289415" iyz="0.0" izz="0.000568125"/>
    </inertial>
  </link>
  <joint name="placer_ee_fixed_joint" type="fixed">
    <parent link="placer_wrist_3_link"/>
    <child link="placer_ee_link"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.0922 0.0"/>
  </joint>
  <link name="placer_ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="placer_shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="placer_shoulder_pan_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="placer_shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="placer_shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="placer_shoulder_lift_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="placer_shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="placer_elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="placer_elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="placer_elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="placer_wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="placer_wrist_1_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="placer_wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="placer_wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="placer_wrist_2_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="placer_wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="placer_wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="placer_wrist_3_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="placer_wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="placer_shoulder_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="placer_upper_arm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="placer_forearm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="placer_wrist_1_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="placer_wrist_3_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="placer_wrist_2_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="placer_ee_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="placer_base"/>
  <joint name="placer_base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="placer_base_link"/>
    <child link="placer_base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="placer_tool0"/>
  <joint name="placer_wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.0922 0"/>
    <parent link="placer_wrist_3_link"/>
    <child link="placer_tool0"/>
  </joint>
  <!-- Default is 0.01 -->
  <!-- 0.0393701 -->
  <!-- Phantom link for center of gripper -->
  <link name="gripper_center">
    <inertial>
      <mass value="0.01"/>
      <inertia ixx="0.0001" ixy="0.0" ixz="0.0" iyy="0.0001" iyz="0.0" izz="0.0001"/>
    </inertial>
  </link>
  <!-- One side of gripper -->
  <link name="side_p">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.11485"/>
      <geometry>
        <mesh filename="package://hydra/models/gripper_placer/meshes/gripper_pad_v2.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.11485"/>
      <geometry>
        <mesh filename="package://hydra/models/gripper_placer/meshes/gripper_pad_v2.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.001"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.02" iyz="0.0" izz="0.02"/>
    </inertial>
  </link>
  <!-- Second side of gripper -->
  <link name="side_d">
    <visual>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0.11485"/>
      <!--origin xyz="0 0.04185 0" rpy="0 0 ${pi}" /-->
      <geometry>
        <mesh filename="package://hydra/models/gripper_placer/meshes/gripper_pad_v2.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </visual>
    <collision>
      <origin rpy="0 0 3.14159265359" xyz="0 0 0.11485"/>
      <geometry>
        <mesh filename="package://hydra/models/gripper_placer/meshes/gripper_pad_v2.stl" scale="0.0254 0.0254 0.0254"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.001"/>
      <inertia ixx="0.01" ixy="0.0" ixz="0.0" iyy="0.02" iyz="0.0" izz="0.02"/>
    </inertial>
  </link>
  <!-- Prismatic joint -->
  <joint name="gripper_joint_p" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="gripper_center"/>
    <child link="side_p"/>
    <limit effort="5000" lower="-100" upper="100" velocity="0.5"/>
  </joint>
  <joint name="gripper_joint_d" type="prismatic">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
    <parent link="gripper_center"/>
    <child link="side_d"/>
    <limit effort="5000" lower="-100" upper="100" velocity="0.5"/>
  </joint>
  <!-- Prismatic joint -->
  <!--joint name="gripper_joint_p" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0" />
    <parent link="gripper_center" />
    <child link="side_p" />
  </joint>
  
  <joint name="gripper_joint_d" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0" />
    <parent link="gripper_center" />
    <child link="side_d" />
    </joint-->
  <transmission name="gripper_joint_transmission_p">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_joint_p">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="gripper_joint_motor_p">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="gripper_joint_transmission_d">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_joint_d">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="gripper_joint_motor_d">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- Gazebo tag for hopefully changing physics engine properties to work -->
  <gazebo>
    <maxVel>0</maxVel>
    <minDepth>0.001</minDepth>
  </gazebo>
  <!-- Attach placer manipulator to world -->
  <link name="world"/>
  <joint name="placer_world_joint" type="fixed">
    <parent link="world"/>
    <child link="placer_base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.0"/>
  </joint>
  <!-- joint to attach mount for tool to UR10 -->
  <joint name="mount_gripper_to_placer" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="placer_tool0"/>
    <child link="gripper_center"/>
  </joint>
  <link name="attacher_base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0061063308908" ixy="0.0" ixz="0.0" iyy="0.0061063308908" iyz="0.0" izz="0.01125"/>
    </inertial>
  </link>
  <joint name="attacher_shoulder_pan_joint" type="revolute">
    <parent link="attacher_base_link"/>
    <child link="attacher_shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1273"/>
    <axis xyz="0 0 1"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="attacher_shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="7.778"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0314743125769" ixy="0.0" ixz="0.0" iyy="0.0314743125769" iyz="0.0" izz="0.021875625"/>
    </inertial>
  </link>
  <joint name="attacher_shoulder_lift_joint" type="revolute">
    <parent link="attacher_shoulder_link"/>
    <child link="attacher_upper_arm_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.220941 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="330.0" lower="-3.14159265359" upper="3.14159265359" velocity="2.16"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="attacher_upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="12.93"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.306"/>
      <inertia ixx="0.421753803798" ixy="0.0" ixz="0.0" iyy="0.421753803798" iyz="0.0" izz="0.036365625"/>
    </inertial>
  </link>
  <joint name="attacher_elbow_joint" type="revolute">
    <parent link="attacher_upper_arm_link"/>
    <child link="attacher_forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.1719 0.612"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.15"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="attacher_forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.87"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.28615"/>
      <inertia ixx="0.111069694097" ixy="0.0" ixz="0.0" iyy="0.111069694097" iyz="0.0" izz="0.010884375"/>
    </inertial>
  </link>
  <joint name="attacher_wrist_1_joint" type="revolute">
    <parent link="attacher_forearm_link"/>
    <child link="attacher_wrist_1_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.5723"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="attacher_wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="attacher_wrist_2_joint" type="revolute">
    <parent link="attacher_wrist_1_link"/>
    <child link="attacher_wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.1149 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="attacher_wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.96"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.0051082479567" ixy="0.0" ixz="0.0" iyy="0.0051082479567" iyz="0.0" izz="0.0055125"/>
    </inertial>
  </link>
  <joint name="attacher_wrist_3_joint" type="revolute">
    <parent link="attacher_wrist_2_link"/>
    <child link="attacher_wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1157"/>
    <axis xyz="0 1 0"/>
    <limit effort="54.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.2"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="attacher_wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_description/meshes/ur10/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.202"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.000526462289415" ixy="0.0" ixz="0.0" iyy="0.000526462289415" iyz="0.0" izz="0.000568125"/>
    </inertial>
  </link>
  <joint name="attacher_ee_fixed_joint" type="fixed">
    <parent link="attacher_wrist_3_link"/>
    <child link="attacher_ee_link"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.0922 0.0"/>
  </joint>
  <link name="attacher_ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="attacher_shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="attacher_shoulder_pan_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="attacher_shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="attacher_shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="attacher_shoulder_lift_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="attacher_shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="attacher_elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="attacher_elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="attacher_elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="attacher_wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="attacher_wrist_1_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="attacher_wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="attacher_wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="attacher_wrist_2_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="attacher_wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="attacher_wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="attacher_wrist_3_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="attacher_wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="attacher_shoulder_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="attacher_upper_arm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="attacher_forearm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="attacher_wrist_1_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="attacher_wrist_3_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="attacher_wrist_2_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="attacher_ee_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="attacher_base"/>
  <joint name="attacher_base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="attacher_base_link"/>
    <child link="attacher_base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="attacher_tool0"/>
  <joint name="attacher_wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.0922 0"/>
    <parent link="attacher_wrist_3_link"/>
    <child link="attacher_tool0"/>
  </joint>
  <joint name="attacher_world_joint" type="fixed">
    <parent link="world"/>
    <child link="attacher_base_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="1.5 0.0 0.0"/>
  </joint>
  <!-- Gazebo plugin for controlling joints -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="ros_control"/>
  </gazebo>
</robot>

